indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorManager.java

```diff
@@ -118,9 +117,24 @@ public void stop()
 
   public Optional<SupervisorReport> getSupervisorStatus(String id)
   {
-    return supervisors.get(id) == null
-           ? Optional.<SupervisorReport>absent()
-           : Optional.fromNullable(supervisors.get(id).lhs.getStatus());
+    Pair<Supervisor, SupervisorSpec> supervisor = supervisors.get(id);
+    return supervisor == null ? Optional.<SupervisorReport>absent() : Optional.fromNullable(supervisor.lhs.getStatus());
+  }
+
+  private boolean possiblyStopAndRemoveSupervisorInternal(String id, boolean writeTombstone)
+  {
+    Pair<Supervisor, SupervisorSpec> pair = supervisors.get(id);
+    if (pair != null) {
+      if (writeTombstone) {
+        metadataSupervisorManager.insert(id, new NoopSupervisorSpec()); // where NoopSupervisorSpec is a tombstone
+      }
+      pair.lhs.stop(true);
+      supervisors.remove(id);
```

Comment:
`io.druid.indexing.overlord.supervisor.Supervisor#stop` doesn't explain expected behavior on stop failure, can you coordinate with @gianm to clarify the behavior of the supervisor `stop` method under the kinds of conditions you mentioned?